lte/gateway/c/core/oai/include/proto_map.hpp

```diff
@@ -0,0 +1,249 @@
+  **                                                                        **
+  ** Description: Returns true if map is empty, else returns false          **
+  **                                                                        **
+  ***************************************************************************/
+  bool isEmpty() { return map->empty(); }
+
+  /***************************************************************************
+  **                                                                        **
+  ** Name:    clear()                                                       **
+  **                                                                        **
+  ** Description: Clears the contents of the map                            **
+  **                                                                        **
+  ***************************************************************************/
+  void clear() {
+    if (!(map->empty())) {
+      map->clear();
+    }
+  }
+  /***************************************************************************
+  **                                                                        **
+  ** Name:    size()                                                        **
+  **                                                                        **
+  ** Description: size the contents of the map                              **
+  **                                                                        **
+  ***************************************************************************/
+  size_t size() { return map->size(); }
+
+  /***************************************************************************
+  **                                                                        **
+  ** Name:    destroy_map()                                                  **
+  **                                                                        **
+  ** Description: Clears the contents of the map and also delete map        **
+  **                                                                        **
+  ***************************************************************************/
+  void destroy_map() {
+    map->clear();
+    delete map;
+  }
+
+  /***************************************************************************
+  **                                                                        **
+  ** Name:    map_apply_callback_on_all_elements()                          **
+  **                                                                        **
+  ** Description: Traverses through map and call callback function to be    **
+  **              executed on each node                                     **
+  **                                                                        **
+  ***************************************************************************/
+  map_apply_callback_on_all_elements(bool funct_cb(const keyT key,
```

Comment:
Can we add a return type here, probably it should be ` proto_map_return_code_e map_apply_callback_on_all_elements(`?